### DISCUSSION TITLE
KernelServicePortalBuilder ("userName","password")

### DISCUSSION THREAD

**Discussion by naiveZhou** [openTCS/opentcs#19] [Q&A] [answered]
Hi，erveryone ,i can't find username and password in  OpenTCS  5.4.
I do not set the username,how can i to connect the OpenTCS kernel from JAVA.
Please let me know if anyone knows, thanks a lot
![image](https://user-images.githubusercontent.com/18117945/169217154-3f9b2fd1-9c9d-40e1-b88b-497408938053.png)

**Comment by martingr** [comment, marked as answer]
Hi,

you can use the credentials defined in `org.opentcs.common.GuestUserCredentials`. These are the ones the kernel uses by default.

**Reply by naiveZhou** [reply]
Thank you so much.
